### PR TITLE
feat: pre-load all relevant workspace files before executing pipeline

### PR DESCRIPTION
**Pull request by WinPlay02**
- relevant workspace files are preloaded and refreshed on each execution
- `executePipeline` function now accepts a `LangiumDocument` instead of a pipeline path
-  fix: filling console causing unresponsiveness when handling new runner output by truncating too long output

### CODE REVIEW

File: packages/safe-ds-vscode/src/extension/pythonServer.ts
Comment (WinPlay02) [inline packages/safe-ds-vscode/src/extension/pythonServer.ts:409]: @SmiteDeluxe According to my tests, this change fixes the unresponsiveness after loading the large table. Please test, if this is the case for you.

File: packages/safe-ds-vscode/src/extension/mainClient.ts
Comment (lars-reimann) [inline packages/safe-ds-vscode/src/extension/mainClient.ts:131]: The function `acceptRunRequests` gets quite long, and it does more than its name implies. Better split this into three separate ones:
1. Register message callbacks
2. Register VS Code commands (maybe even extract the callback for each command into a named function)
3. Register watchers

Later we can move parts of this into separate files.

File: packages/safe-ds-vscode/src/extension/mainClient.ts
Comment (lars-reimann) [inline packages/safe-ds-vscode/src/extension/mainClient.ts:182]:
```suggestion
                if (!services.shared.workspace.LangiumDocuments.hasDocument(pipelinePath)) {
```

Doesn't this work too?

**Review by lars-reimann — CHANGES_REQUESTED** [review]

**Review by lars-reimann — APPROVED** [review]
Looks good. Let's wait for @SmiteDeluxe's response before merging, though.

File: packages/safe-ds-vscode/src/extension/pythonServer.ts
Comment (lars-reimann) [reply packages/safe-ds-vscode/src/extension/pythonServer.ts:420]: I'll merge it now, we've waited long enough for a response.

**Pull request merged**
